### PR TITLE
Teleportation wand artifact effect

### PR DESCRIPTION
**Pull request by Boaz1111**
## About the PR

added a teleportation wand artifact effect, as well as an instant scan debug artifact analyser(mapped in the devenvironment)
## Why / Balance

More utility arti effects, and I found myself making great use of the scanner while testing. 
## Technical details

removed the trigger artifact verb from debug, for ease of access
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added a new "teleporation wand" artifact effect

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Specific/Janitorial/spray.yml [hunk 1]
```diff
@@ -147,3 +147,28 @@
         mask:
         - FullTileMask
         - Opaque
+
+- type: entity
+  name: mega spray bottle
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Specific/Janitorial/spray.yml:152]: remove dupe name desc sprite and maybe spray

File: Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml [hunk 2]
```diff
@@ -211,3 +211,16 @@
       machine_parts: !type:Container
       entity_storage: !type:Container
       output_container: !type:Container
+
+- type: entity
+  id: MachineArtifactAnalyzerDebug
+  parent: MachineArtifactAnalyzer
+  name: artifact analyzer
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml:218]: remove name and desc

File: Resources/Prototypes/XenoArch/Effects/utility_effects.yml [hunk 3]
```diff
@@ -247,3 +247,18 @@
...
+
+- type: artifactEffect
+  id: EffectDash
+  targetDepth: 3
+  effectHint: artifact-effect-hint-displacement
+  whitelist:
+    components:
+    - Item
+  permanentComponents:
+  - type: DashAbility
```
Comment (deltanedas) [inline Resources/Prototypes/XenoArch/Effects/utility_effects.yml:259]: fairly sure this requires ninja to be used?

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
debug artifexium shouldnt be necessary you can activate artifacts in debug with a verb anyway

File: Resources/Prototypes/XenoArch/Effects/utility_effects.yml (hunk 3 above)
Comment (Boaz1111) [reply Resources/Prototypes/XenoArch/Effects/utility_effects.yml:259]: it doesn't, I tested

**Comment by icekot8** [comment]
1. I don't like that you are trying to add 3 almost unrelated things through 1 PR
2. I didn't quite understand from your description what your effect does. From what I understand, it turns the artifact into a teleportation magic wand (currently debug), with 1 charge every 15 seconds. I think that's a bit too strong. Also, I don't think your effect would mix well with the artifact-gun effect.

**Comment by Hent798** [comment]
It's too unbalanced for an artifact, I believe it will be passive like a revolver, which makes this wand too easy to use.
I could understand if teleportation was tied to one specific node, which would make it difficult to use and require certain skills and experience with artifacts.

Now you just want to give the artifact a completely unbalanced effect, but you don't want to put any effort into using it.

**Comment by Boaz1111** [comment]
> 1. I don't like that you are trying to add 3 almost unrelated things through 1 PR
> 
> 2. I didn't quite understand from your description what your effect does. From what I understand, it turns the artifact into a teleportation magic wand (currently debug), with 1 charge every 15 seconds. I think that's a bit too strong. Also, I don't think your effect would mix well with the artifact-gun effect.

It's not a debug thing, ignore that. You know the energy katana? The warp ability? It gives that, but with a max single charge and a 15(will bump to 30) second cooldown

**Comment by foboscheshir** [comment]
OP 💀

**Comment by Hent798** [comment]
even with a 1 minute cooldown it's too unbalanced
such a teleport would look at least somewhat acceptable as a reaction of one specific node in order to move to the next node after activation

Moreover, as a reaction of one node, this effect can synergize with other nodes.

**Pull request closed without merging**